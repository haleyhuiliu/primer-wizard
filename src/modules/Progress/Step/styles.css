.step {
  color: #888;
  float: left;
  padding: 10px 5px 10px 30px;
  position: relative;
}

.step::before,
.step::after {
  border-left: solid 20px #3B5998;
  border-bottom: solid 27px transparent;
  border-top: solid 27px transparent;
  content: '';
  display: block;
  height: 0;
  position: absolute;
  top: 50%;
  transform: translateY(-50%);
  width: 0;
}

.step::before {
  left: 0;
}

.step::after {
  left: 100%;
}

.step.active::before {
  border-left-color: #DFE3EE;
}

.step.active::after {
  border-left-color: #3B5998;
}

.step:first-of-type::before {
  border-left-color: transparent;
}

.step:not(.active)::after {
  border-left-color: #DFE3EE;
}

.step:nth-of-type(n - 1):not(.active)::before {
  border-left-color: transparent;
}

.active {
  background: #3B5998;
  color: #fff;
}

.step-title {
  font-weight: 800;
  margin-bottom: 4px;
}

.step-subtitle {
  font-size: 0.875em;
}
