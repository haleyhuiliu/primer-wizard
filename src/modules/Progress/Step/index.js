import React, { PropTypes } from 'react';

import styles from './styles.css';

const Step = props => {
  const { title, subtitle, active } = props;
  return (
    <div className={`${styles.step} ${active ? styles.active : ''}`}>
      <div className={styles['step-title']}>{title}</div>
      <div className={styles['step-subtitle']}>{subtitle}</div>
    </div>
  );
};

Step.propTypes = {
  title: PropTypes.oneOfType([PropTypes.string, PropTypes.node]),
  subtitle: PropTypes.oneOfType([PropTypes.string, PropTypes.node]),
  active: PropTypes.bool,
};

export default Step;
