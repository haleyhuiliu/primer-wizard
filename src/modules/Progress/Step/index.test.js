import React from 'react';
import renderer from 'react-test-renderer';

import Step from '../Step';

it('renders a Step component', () => {
  const props = {
    title: 'Sample Test',
    subtitle: 'Sample Subtitle',
    active: false,
  };

  const tree = renderer.create(<Step {...props} />);
  expect(tree).toMatchSnapshot();
});
