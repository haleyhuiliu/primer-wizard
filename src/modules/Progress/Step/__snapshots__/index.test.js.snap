// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a Step component 1`] = `
<div
  className="step "
>
  <div
    className="step-title"
  >
    Sample Test
  </div>
  <div
    className="step-subtitle"
  >
    Sample Subtitle
  </div>
</div>
`;
